File: sdks/rust/src/harnessaas/envelope.rs
//! Result and metadata envelope for HarnessaaSClient operations (ADR-0027a).

/// Per-response metadata carried alongside every [`HarnessaaSResult`].
#[derive(Debug, Clone)]
pub struct HarnessaaSResponseMeta {
    pub request_id: String,
    pub http_status: u16,
    pub retry_after_ms: Option<u64>,
}

/// Envelope wrapping every HarnessaaSClient operation result.
#[derive(Debug, Clone)]
pub struct HarnessaaSResult<T> {
    pub data: T,
    pub meta: HarnessaaSResponseMeta,
}

File: sdks/node/src/harnessaas/envelope.ts
/** Result and metadata envelope for HarnessaaSClient operations (ADR-0027a). */

/** Per-response metadata carried alongside every {@link HarnessaaSResult}. */
export interface HarnessaaSResponseMeta {
  requestId: string;
  httpStatus: number;
  retryAfterMs?: number;
}

/** Envelope wrapping every HarnessaaSClient operation result. */
export interface HarnessaaSResult<T> {
  data: T;
  meta: HarnessaaSResponseMeta;
}
